packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-group.ts

```diff
@@ -32,14 +38,151 @@ export interface IEcsDeploymentGroup extends cdk.IResource {
+   *
+   * If a test listener is not specified, the deployment will proceed to routing the production listener to the 'green' ECS task set
+   * and will skip the AfterAllowTestTraffic hook.
+   *
+   * @default No test listener will be added
+   */
+  readonly testListener?: elbv2.IListener;
+
+  /**
+   * Specify how long CodeDeploy waits for approval to continue a blue-green deployment before it stops the deployment.
+   *
+   * After provisioning the 'green' ECS task set and re-routing test traffic, CodeDeploy can wait for approval before
+   * continuing the deployment and re-routing production traffic.  During this wait time, validation such as manual
+   * testing or running integration tests can occur using the test traffic port, prior to exposing the new 'green' task
+   * set to production traffic.  To approve the deployment, validation steps use the CodeDeploy
+   * [ContinueDeployment API(https://docs.aws.amazon.com/codedeploy/latest/APIReference/API_ContinueDeployment.html).
+   * If the ContinueDeployment API is not called within the wait time period, CodeDeploy will stop the deployment.
+   *
+   * By default, CodeDeploy will not wait for deployment approval.  After re-routing test traffic to the 'green' ECS task set
+   * and running any 'AfterAllowTestTraffic' and 'BeforeAllowTraffic' lifecycle hooks, the deployment will immediately
+   * re-route production traffic to the 'green' ECS task set.
+   *
+   * @default 0
+   */
+  readonly deploymentApprovalWaitTime?: cdk.Duration;
+
+  /**
+    * Specify how long CodeDeploy waits before it terminates the original 'blue' ECS task set when a blue-green deployment is complete.
+    *
+    * During this wait time, CodeDeploy will continue to monitor any CloudWatch alarms specified for the deployment group,
+    * and the deployment group can be configured to automatically roll back if those alarms fire.  Once CodeDeploy begins to
+    * terminate the 'blue' ECS task set, the deployment can no longer be rolled back, manually or automatically.
+    *
+    * By default, the deployment will immediately terminate the 'blue' ECS task set after production traffic is successfully
+    * routed to the 'green' ECS task set.
+    *
+    * @default 0
+    */
+  readonly terminationWaitTime?: cdk.Duration;
+}
+
+/**
+ * Construction properties for {@link EcsDeploymentGroup}.
+ */
+export interface EcsDeploymentGroupProps {
+  /**
+   * The reference to the CodeDeploy ECS Application that this Deployment Group belongs to.
+   *
+   * @default One will be created for you.
+   */
+  readonly application?: IEcsApplication;
+
+  /**
+   * The physical, human-readable name of the CodeDeploy Deployment Group.
+   *
+   * @default An auto-generated name will be used.
+   */
+  readonly deploymentGroupName?: string;
+
+  /**
+   * The Deployment Configuration this Deployment Group uses.
+   *
+   * @default EcsDeploymentConfig.ALL_AT_ONCE
+   */
+  readonly deploymentConfig?: IEcsDeploymentConfig;
```

Comment:
I think this will need to be updated after #22567

Reply:
Does it?  Unless I'm missing something, the latest changes merged in the PR looked like `IEcsDeploymentConfig` still gets returned by `EcsDeploymentConfig.fromEcsDeploymentConfigName` and `new EcsDeploymentConfig`, so the caller should have that type of object.

Reply:
ah nevermind, I thought they were going to do something different.